src/Lender.sol

```diff
@@ -546,45 +598,43 @@ contract Lender {
         );
 
         // Convert decimals from principal token to underlying
-        uint256 returned = convertDecimals(u, pt, obtained);
+        uint256 returned = convertDecimals(u, _Market.tokens[p], obtained);
 
         // Mint Illuminate PTs to msg.sender
         IERC5095(principalToken(u, m)).authMint(msg.sender, returned);
         emit Lend(p, u, m, returned, spent, msg.sender);
-        return returned;
+        return (returned);
     }
 
-    // @notice: Handles all lending to ETH markets and any swaps necessary
-    // @param adapter: The adapter contract to lend through
+    // @notice: Adjusts all Swivel Amounts according to the slippageRatio
+    // @param slippageRatio: The slippageRatio to adjust by
     // @param d: The calldata for the lend (and potentially swap) operation
-    function ETHLend(address adapter, bytes calldata d) internal returns (bool success, bytes memory returndata) {
-        // Parse the calldata
-        (
-            address underlying,
-            uint256 maturity,
-            address pool,
-            uint256 amount,
-            uint256 minimum,
-            address lst,
-            uint256 swapMinimum
-        ) = abi.decode(d, (address, uint256, address, uint256, uint256, address, uint256));
+    // @returns The adjusted amount array
+    function adjustSwivelAmounts(uint256[] memory amounts, uint256 slippageRatio) internal pure returns (uint256[] memory) {
```

Comment:
this is fine if we'll be adjusting just for Swivel.. if there's a need to adjust amounts for different protocols, we could create a generic function and move the logic to the appropriate adapter if it's not too expensive

Reply:
Yeah the other ones we adjust by just using the return value from the curve swap

Swivel is just unique in that it doesnt have one order amount, and instead has an array that need to be adjusted so you cant use that single amount value

Reply:
👍